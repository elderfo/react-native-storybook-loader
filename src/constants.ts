export const appName = 'react-native-storybook-loader';
export const encoding = 'utf-8';
