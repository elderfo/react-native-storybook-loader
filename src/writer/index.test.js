import mock from 'mock-fs';
import path from 'path';
import fs from 'fs';
import { writeFile } from './';
import { encoding } from '../constants';

const baseDir = path.resolve(__dirname, '../../');
const outputPath = path.resolve(baseDir, 'output.js');
const moduleDir = path.resolve(baseDir);
const configFile = path.resolve(moduleDir, './module.config.json');

beforeEach(() => {
  mock({
    [outputPath]: '',
    [configFile]: '',
  });
});

afterEach(() => {
  mock.restore();
});

test('writeFile should perform expected work', () => {
  const files = [
    path.resolve(__dirname, '../file1.js'),
    path.resolve(__dirname, '../sub/file2.js'),
    path.resolve(__dirname, '../../parent/file3.js'),
    path.resolve(__dirname, './sub/file4.js'),
    path.resolve(__dirname, './sub/sub/file5.js'),
  ];
  writeFile(files, outputPath, moduleDir);

  const contents = fs.readFileSync(outputPath, encoding);

  expect(contents).toMatchSnapshot();

  const configContents = fs.readFileSync(configFile, encoding);
  expect(configContents).toMatchSnapshot();
});
