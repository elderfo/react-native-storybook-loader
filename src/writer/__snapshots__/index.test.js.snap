// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`writeFile should perform expected work 1`] = `
"
// Auto-generated file created by react-native-storybook-loader
// Do not edit.
//
// https://github.com/elderfo/react-native-storybook-loader.git

function loadStories() {
  require('./parent/file3');
  require('./src/file1');
  require('./src/sub/file2');
  require('./src/writer/sub/file4');
  require('./src/writer/sub/sub/file5');
  
}

const stories = [
  './parent/file3',
  './src/file1',
  './src/sub/file2',
  './src/writer/sub/file4',
  './src/writer/sub/sub/file5',
  
];

module.exports = {
  loadStories,
  stories,
};
"
`;
